src/interfaces/swig/GaussianProcess.i

```diff
@@ -6,102 +6,24 @@
-
-%shared_ptr(shogun::KLInference)
-%shared_ptr(shogun::KLLowerTriangularInference)
-%shared_ptr(shogun::KLCovarianceInferenceMethod)
-%shared_ptr(shogun::KLDiagonalInferenceMethod)
-%shared_ptr(shogun::KLCholeskyInferenceMethod)
-%shared_ptr(shogun::KLDualInferenceMethod)
-
+%shared_ptr(shogun::GaussianProcess)
 %shared_ptr(shogun::KLDualInferenceMethodMinimizer)
 
-%shared_ptr(shogun::GaussianProcessMachine)
-%shared_ptr(shogun::GaussianProcessClassification)
-%shared_ptr(shogun::GaussianProcessRegression)
-
 
 /* These functions return new Objects */
 
 /* Include Class Headers to make them visible from within the target language */
-%include <shogun/evaluation/DifferentiableFunction.h>
 %include <shogun/machine/gp/LikelihoodModel.h>
 RANDOM_INTERFACE(LikelihoodModel)
-%include <shogun/machine/gp/ProbitLikelihood.h>
-%include <shogun/machine/gp/LogitLikelihood.h>
-%include <shogun/machine/gp/SoftMaxLikelihood.h>
-%include <shogun/machine/gp/GaussianLikelihood.h>
-%include <shogun/machine/gp/StudentsTLikelihood.h>
-
-%include <shogun/machine/gp/VariationalLikelihood.h>
-%include <shogun/machine/gp/VariationalGaussianLikelihood.h>
-%include <shogun/machine/gp/NumericalVGLikelihood.h>
-%include <shogun/machine/gp/DualVariationalGaussianLikelihood.h>
-%include <shogun/machine/gp/LogitVGLikelihood.h>
-%include <shogun/machine/gp/LogitVGPiecewiseBoundLikelihood.h>
-%include <shogun/machine/gp/LogitDVGLikelihood.h>
-%include <shogun/machine/gp/ProbitVGLikelihood.h>
-%include <shogun/machine/gp/StudentsTVGLikelihood.h>
 
 %include <shogun/machine/gp/MeanFunction.h>
-%include <shogun/machine/gp/ZeroMean.h>
-%include <shogun/machine/gp/ConstMean.h>
+
 
 %include <shogun/machine/gp/Inference.h>
 RANDOM_INTERFACE(Inference)
-%include <shogun/machine/gp/LaplaceInference.h>
-%include <shogun/machine/gp/SparseInference.h>
-%include <shogun/machine/gp/SingleSparseInference.h>
-%include <shogun/machine/gp/SingleFITCInference.h>
-%include <shogun/machine/gp/SingleLaplaceInferenceMethod.h>
-%include <shogun/machine/gp/MultiLaplaceInferenceMethod.h>
-%include <shogun/machine/gp/ExactInferenceMethod.h>
-%include <shogun/machine/gp/SingleFITCLaplaceInferenceMethod.h>
-%include <shogun/machine/gp/FITCInferenceMethod.h>
-%include <shogun/machine/gp/VarDTCInferenceMethod.h>
-%include <shogun/machine/gp/EPInferenceMethod.h>
-
-%include <shogun/machine/gp/KLInference.h>
-%include <shogun/machine/gp/KLLowerTriangularInference.h>
-%include <shogun/machine/gp/KLCovarianceInferenceMethod.h>
-%include <shogun/machine/gp/KLDiagonalInferenceMethod.h>
-%include <shogun/machine/gp/KLCholeskyInferenceMethod.h>
-%include <shogun/machine/gp/KLDualInferenceMethod.h>
-
-%include <shogun/machine/GaussianProcessMachine.h>
-%include <shogun/classifier/GaussianProcessClassification.h>
-%include <shogun/regression/GaussianProcessRegression.h>
+%include <shogun/machine/GaussianProcess.h>
+%include <shogun/machine/gp/KLDualInferenceMethod.h> //KLDualInferenceMethodMinimizer
```

Comment:
this one stays?

Reply:
notebooks use the ` KLDualInferenceMethodMinimizer`.
https://github.com/shogun-toolbox/shogun/blob/9b5793b6dff592a9caac3b41f5359838e3ac2ce4/src/shogun/machine/gp/KLDualInferenceMethod.h#L48-L90
should i put the  ` KLDualInferenceMethodMinimizer` to other places?

Reply:
Shouldn’t minimisers be exposed to swig?

Reply:
the `KLDualInferenceMethodMinimizer` should be exposed to swig IMO

Reply:
shouldn't it be available through `sg.create_minimizer("KLDualInferenceMethodMinimizer")`?

Reply:
it seems like that we don't have the factory for `Minimizer`.

Reply:
We don't but should be easy to add no?

Reply:
@karlnapf I am assuming that would be another class we would to instantiate through factories right?

Reply:
there is no method in `Minimizer`, maybe we need to refactor the `Minimizer` to make factory work.
https://github.com/shogun-toolbox/shogun/blob/9b5793b6dff592a9caac3b41f5359838e3ac2ce4/src/shogun/optimization/Minimizer.h#L43-L53

Reply:
not sure what you mean with there is no method?

Reply:
sorry, i wrongly think `virtual float64_t minimize()=0; ` is constructor, you are right, it should be easy to add the factory, will do :)

Reply:
Ah I see, no problem! Thanks :)